Recent change to SetupCoin3D.cmake breaks snap package build

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

As reported in the forum: https://forum.freecad.org/viewtopic.php?p=767900#p767900

Since two days the daily packages have failed to build: https://github.com/FreeCAD/FreeCAD-snap/actions.

At a quick look, it seems there is an issue importing the 'pivy' module: https://github.com/FreeCAD/FreeCAD-snap/actions/runs/9654503841/job/26628851594#step:4:169

```
:: -- Checking Pivy version by importing it in a Python program...
:: Traceback (most recent call last):
::   File "<string>", line 1, in <module>
:: ModuleNotFoundError: No module named 'pivy'
:: CMake Error at cMake/FreeCAD_Helpers/SetupCoin3D.cmake:40 (message):
::   Failed to import Pivy using /usr/bin/python3.10
:: Call Stack (most recent call first):
::   CMakeLists.txt:86 (SetupCoin3D)
::
::
:: -- Configuring incomplete, errors occurred!
:: error:
:: Traceback (most recent call last):
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 111, in main
::     ret = CraftCtl.run(cmd, args)
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 45, in run
::     _client(cmd, args)
::   File "/snap/snapcraft/current/lib/python3.10/site-packages/craft_parts/ctl.py", line 98, in _client
::     raise RuntimeError(message)
:: RuntimeError: Failed to run the build script for part 'freecad'.
'override-build' in part 'freecad' failed with code 1.
Review the scriptlet and make sure it's correct.
Full execution log: '/root/.local/state/snapcraft/log/snapcraft-20240625-004209.501939.log'
Error: Process completed with exit code 1.
```

Then looking at the last successful build: https://github.com/FreeCAD/FreeCAD-snap/actions/runs/9637706423/job/26577337297#step:4:170 I noticed that Pivy fails to be imported too, but it does not result in a build error:

```python-traceback
 ModuleNotFoundError: No module named 'pivy'
```

It also looks like this PR, coincidentally merged 2 days ago too, touches the file that is generating the build error ([cMake/FreeCAD_Helpers/SetupCoin3D.cmake](https://github.com/FreeCAD/FreeCAD/commit/d01ce32a1f6de77a6b3438413d9a35137061ff8f#diff-02b39c479ae55956e97ee61ae91204e885aa8471c1815d77ecaf0eb625f2d0c1)): https://github.com/FreeCAD/FreeCAD/pull/10337

**Before #10337**:
https://github.com/FreeCAD/FreeCAD/blob/44e8f910852101aba05853ec37315c56c641e18e/cMake/FreeCAD_Helpers/SetupCoin3D.cmake#L35-L37

**After #10337**:
https://github.com/FreeCAD/FreeCAD/blob/d01ce32a1f6de77a6b3438413d9a35137061ff8f/cMake/FreeCAD_Helpers/SetupCoin3D.cmake#L31-L42

I believe that the issue is that in the snap build the Pivy import had always failed (for whichever reason), then the failure was ignored and the build continued. The latest change on #10337 (line 40 in the snippet above) makes the failure a fatal error, so the build stops.

### Full version info

```shell
OS: Ubuntu Core 22 (ubuntu:GNOME/ubuntu)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37860 (Git) Snap 1060
Build type: Release
Branch: main
Hash: d2491541e19085aa47ea6662bcb7a3da6b66770c
Python 3.10.12, Qt 5.15.10, Coin 4.0.0, Vtk 7.1.1, OCC 7.7.1
Installed mods: 
  * OpenTheme 2024.5.16
  * Curves 0.6.39
  * ezydxf
  * ExplodedAssembly
  * BillOfMaterials 0.0.8.5
  * Ondsel-Lens 2024.5.6.01
  * Assembly4 0.50.13
```


### Subproject(s) affected?

Other (specify in description)

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/cMake/FreeCAD_Helpers/SetupCoin3D.cmake
+++ b/cMake/FreeCAD_Helpers/SetupCoin3D.cmake
@@ -37,7 +37,7 @@ macro(SetupCoin3D)
         if (RETURN_CODE EQUAL 0)
             message(STATUS "Found Pivy ${PIVY_VERSION}")
         else ()
-            message(FATAL_ERROR "Failed to import Pivy using ${Python3_EXECUTABLE}")
+            message(ERROR "Failed to import Pivy using ${Python3_EXECUTABLE}")
         endif ()
     ENDIF ()
 

